lnc/lnc.go

```diff
@@ -0,0 +1,169 @@
+package lnc
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightninglabs/lightning-node-connect/mailbox"
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/mwitkow/grpc-proxy/proxy"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+// HeaderMacaroon is the HTTP header field name that is used to send
+// the macaroon.
+const HeaderMacaroon = "Macaroon"
+
+type conn struct {
+	client     lnrpc.LightningClient
+	grpcClient *grpc.ClientConn
+	creds      credentials.PerRPCCredentials
+	cancel     func()
+}
+
+func (c *conn) Close() error {
+	err := c.grpcClient.Close()
+	c.cancel()
+	return err
+}
+
+type LNC struct {
+	store Store
+
+	passphrase []byte
+	conn       *conn
+	mu         sync.Mutex
+}
+
+func NewLNC(store Store) *LNC {
+	return &LNC{
+		store: store,
+		mu:    sync.Mutex{},
+	}
+}
+
+func (l *LNC) OpenConn(ctx context.Context, passphrase []byte) error {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	if l.conn != nil {
+		return fmt.Errorf("connection already open")
+	}
+
+	session, err := l.store.GetSessionByPassphrase(ctx, passphrase)
+	if err != nil {
+		return err
+	}
+
+	conn, err := l.newConn(session)
+	if err != nil {
+		return err
+	}
+
+	l.conn = conn
+	l.passphrase = passphrase
+
+	return nil
+}
+
+func (l *LNC) CloseConn() error {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	if l.conn == nil {
+		return fmt.Errorf("connection not open")
+	}
+
+	return l.conn.Close()
+}
+
+func (l *LNC) Stop() error {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	if l.conn != nil {
+		return l.CloseConn()
+	}
+
+	return nil
+}
+
+func (l *LNC) onRemoteStatic(key *btcec.PublicKey) error {
+	ctxt, cancel := context.WithTimeout(
+		context.Background(), time.Second*10,
+	)
+	defer cancel()
+
+	remoteKey := key.SerializeCompressed()
+
+	return l.store.SetRemotePubKey(ctxt, l.passphrase, remoteKey)
+}
+
+func (l *LNC) onAuthData(data []byte) error {
+	return nil
```

Comment:
IIRC, this is the call back that'll be used to get the macaroon in the latter portion of the first handshake? 

cc @ellemouton @guggero

Reply:
yeah